test/opt/test_set_spec_const_default_value.cpp

```diff
@@ -20,6 +20,120 @@ using namespace spvtools;
 using SpecIdToValueStrMap =
     opt::SetSpecConstantDefaultValuePass::SpecIdToValueStrMap;
 
+struct DefaultValuesStringParsingTestCase {
+  const char* default_values_str;
+  bool expect_success;
+  SpecIdToValueStrMap expected_map;
+};
+
+using DefaultValuesStringParsingTest =
+    ::testing::TestWithParam<DefaultValuesStringParsingTestCase>;
+
+TEST_P(DefaultValuesStringParsingTest, TestCase) {
+  // const auto& tc = GetParam();
+  // SinglePassRunAndCheck<opt::SetSpecConstantDefaultValuePass>(
+  // tc.code, tc.expected, [> skip_nop = <] false, tc.default_values);
+  const auto& tc = GetParam();
+  auto actual_map =
+      opt::SetSpecConstantDefaultValuePass::ParseDefaultValuesString(
+          tc.default_values_str);
+  if (tc.expect_success) {
+    EXPECT_NE(nullptr, actual_map);
+    EXPECT_EQ(tc.expected_map, *actual_map.get());
+  } else {
+    EXPECT_EQ(nullptr, actual_map);
+  }
+}
+
+INSTANTIATE_TEST_CASE_P(
+    ValidString, DefaultValuesStringParsingTest,
+    ::testing::ValuesIn(std::vector<DefaultValuesStringParsingTestCase>{
+        // 0. empty map
+        {"{}", true, SpecIdToValueStrMap{}},
+        // 1. one pair
+        {"{100:1024}", true, SpecIdToValueStrMap{{100, "1024"}}},
+        // 2. two pairs
+        {"{100:1024, 200:2048}", true,
+         SpecIdToValueStrMap{{100, "1024"}, {200, "2048"}}},
+        // 3. spaces before colon
+        {"{100   :1024}", true, SpecIdToValueStrMap{{100, "1024"}}},
+        // 4. spaces after colon
+        {"{100:   1024}", true, SpecIdToValueStrMap{{100, "1024"}}},
+        // 5. \t, \n, \r and spaces before and after colon
+        {"{100   \n \r\t \t  : \r \t  \n1024}", true,
+         SpecIdToValueStrMap{{100, "1024"}}},
+        // 6. \t, \n, \r and spaces before spec id
+        {"{   \n \r\t \t  100:1024}", true, SpecIdToValueStrMap{{100, "1024"}}},
+        // 7. \t, \n, \r and spaces after value string
+        {"{100:1024   \n \r\t \t  }", true, SpecIdToValueStrMap{{100, "1024"}}},
+        // 8. maximum spec id
+        {"{4294967295:0}", true, SpecIdToValueStrMap{{4294967295, "0"}}},
+        // 9. minimum spec id
+        {"{0:100}", true, SpecIdToValueStrMap{{0, "100"}}},
+        // 10. random content without spaces are allowed
+        {"{200: random_stuff}", true,
+         SpecIdToValueStrMap{{200, "random_stuff"}}},
+        // 11. missing ending '}' is fine for parsing
+        {"{100:1024", true, SpecIdToValueStrMap{{100, "1024"}}},
+        // 12. multiple ending '}' is fine for parsing
+        {"{100:1024}}", true, SpecIdToValueStrMap{{100, "1024"}}},
```

Comment:
Hmm, currently all the content after the first '}' is ignored. So it doesn't matter whether they are anything after '}' at all. But anyway, this can be added later.